compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -43,19 +43,44 @@ using mlir::iree_compiler::IREE::LinalgExt::LinalgVectorLoweringOptions;
 // IREE specific functions
 //===----------------------------------------------------------------------===//
 
-/// Returns the op that contains lowering config. Returns failure if there are
-/// multiple op having lowering config.
-static FailureOr<Operation *> getRootOp(func::FuncOp funcOp) {
+/// Returns the op that contains lowering config. Checks whether the provided op
+/// contains the lowering config and returns it. Otherwise, tries to find the
+/// lowering config across the function. If there are multiple ops with the same
+/// lowering configs, returns the first one found. Returns failure if there are
+/// multiple op with different lowering config.
```

Comment:
What was the need to change the semantics of this function? Do we have multiple lowering configs now?

Reply:
We can have multiple lowering configs for example after peeling, IIRC. Changes here are extending the API and returning the lowering config when all the lowering config found are the same. The utility was bailing out when multiple lowering configs were found without checking if they were the same.

Reply:
Perhaps add the fact that we can have multiple lowering configs because of peeling?